### ISSUE TITLE
Add support for Matlab style doctests

### ISSUE THREAD

**Issue by skycaptain** [sphinx-contrib/matlabdomain#94]
We're using Sphinx for generating HTML docs out of help texts. We would like to use [catch22/octave-doctest](https://github.com/catch22/octave-doctest) for running tests on the examples in docstrings. Sphinx recognizes doctests starting with `>>> ` and formats them as accordingly. *octave-doctest* however requires doctests starting with `>> `. See discussion here https://github.com/catch22/octave-doctest/pull/236.

This would allow us to use:

```matlab
function number = getMyNumber()
% Returns the number
%
%   Returns:
%       number (numeric): The number
%
%   Example:
%       >> getMyNumber()
%       5
%
    number = 5;
end
```

Instead of:

```matlab
function number = getMyNumber()
% Returns the number
%
%   Returns:
%       number (numeric): The number
%
%   Example:
%       ::
%
%           >> getMyNumber()
%           5
%
    number = 5;
end
```

Not an elegant solution, but one could Monkey-patch [this](https://sources.debian.org/src/python-docutils/0.15.2+dfsg-1/docutils/parsers/rst/states.py/#L1142) line.

**Comment by joeced** [comment]
Thanks for the suggestion. I think it's a nice idea and I tried out your monkey-patching idea, which works as expected (and is easy to implement).

I think I will add a new option, ``matlab_enable_doctest``, which will do the monkey-patching. Does that sound OK?

**Comment by cbm755** [comment]
Julia uses a prompt `julia>`: how do they deal with this in [their doctests](https://juliadocs.github.io/Documenter.jl/stable/man/doctests/#REPL-Examples-1)?

Should we try to get it customizable in Sphinx itself?

**Comment by joeced** [comment]
@cbm755 That might difficult. The problem is that the solution is to monkey-patch ``docutils``, which is not part of Sphinx directly.

**Comment by cbm755** [comment]
Thanks, I see.

This is mentioned under http://docutils.sourceforge.net/docs/dev/todo.html "Generalize the doctest block construct (which is overly Python-centric) to other...".  But at least one person "spoke out against this idea".

So I guess a local fix here is certainly quicker.

**Comment by skycaptain** [comment]
Nice one! Just one more thing, that came to my mind: How does this affect mixed environments, where we have Python and Matlab code? The patch is global I guess and not per-file?

**Comment by skycaptain** [comment]
Another thing that came to my mind: Does Sphinx break the highlighting after one empty line. See the discussion we had over at https://github.com/catch22/octave-doctest/issues/239

**Comment by cbm755** [comment]
The [to-be-written manual for octave-python](https://gitlab.com/mtmiller/octave-pythonic/issues/41) might want to mix Python and Octave doctests in the same document.  CC @mtmiller.

I'm not a Sphinx/Docutils user yet so maybe this is not useful...  Can we write some kind of thing like https://www.sympy.org/sphinx-math-dollar/ but that munges `>>`-style doctests into whatever format Docutils wants?  (i.e., with `<BLANKLINE>` and whatever conventions).

I'd be happy to collaborate on the parsing of octave-doctest Doctests if that were helpful---i.e., we could factor that out a bit.

**Comment by joeced** [comment]
I'll try to answer the comments.

- Unfortunately autodoc with matlab and python does not play well as reported in #63. 
- Sphinx breaks the highlighting after one blank line. 
![image](https://user-images.githubusercontent.com/22655178/66697395-8d0adf80-ecd5-11e9-8ce7-f10a482ad36a.png)
- The idea from @cbm755 might work better, although it requires one to make yet another module.

I pushed the first *hacked* version to https://github.com/sphinx-contrib/matlabdomain/tree/doctests so you can try it out.

You can install it locally (preferably in a virtual environment) with 
```
pip install git+https://github.com/sphinx-contrib/matlabdomain.git/@doctests
```

**Comment by joeced** [comment]
I just released a new version, 0.10.0, where Python and MATLAB code can be documented together (closed #63), which probably makes it a lot more difficult to use the proposed hack.

Another solution could be to connect to the [``autodoc-process-docstring``](https://github.com/sphinx-doc/sphinx/blob/a6a2a59ab0abfaf510d3dcbc43231468d3dc1fff/sphinx/ext/napoleon/__init__.py#L304) as Napoleon extension does and scan for ``>>`` or whatever doctests symbol should be used.

**Comment by joeced** [comment]
I've decided not to support this feature.
